src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
 
     return rc;
 }
+
+#ifdef HDF5_SUPPORT
+static inline void init_hdf5_args(struct hdf5_args *hdf5)
+{
+    hdf5->status = 0;
+    hdf5->file = -1;
+    /* OID Data */
+    hdf5->oid_memtype = 0;
+    hdf5->oid_dspace = 0;
+    hdf5->oid_dset = 0;
+    /* DKEY Data */
+    hdf5->dkey_memtype = 0;
+    hdf5->dkey_vtype = 0;
+    hdf5->dkey_dspace = 0;
+    hdf5->dkey_dset = 0;
+    /* AKEY Data */
+    hdf5->akey_memtype = 0;
+    hdf5->akey_vtype = 0;
+    hdf5->akey_dspace = 0;
+    hdf5->akey_dset = 0;
+    /* dims for dsets */
+    hdf5->oid_dims[0] = 0;
+    hdf5->dkey_dims[0] = 0;     
+    hdf5->akey_dims[0] = 0;     
+    /* data for keys */
+    hdf5->oid_data = NULL;
+    hdf5->dkey_data = NULL;
+    hdf5->akey_data = NULL;
+    hdf5->dk = NULL;
+    hdf5->ak = NULL;
+}
+
+static int serialize_recx_single(struct hdf5_args *hdf5, 
```

Comment:
@dsikich There is some overlap between these iteration functions and the ones for dcp/dsync.
When I refactor the functions for dcp/dsync like we talked about before, I'll see if this can fit in as well.